### PR TITLE
Remove Manual Groups

### PR DESCRIPTION
**Pull request by evantahler**
To simplify & clarify Grouparoo, this PR removes the concept of "manual" Groups.  From this point forward, all Grouparoo Groups are "calculated", meaning that they dynamically determine which Records are members based on the Properties of those Records.  This has the effect of simplifying many things from the UI, configuration files, and internal logic.  This is a breaking change.

After the application of this PR, the following breaking changes will happen:
* The migration will fail if there are manual Groups in the database.  These will need to be removed before the migration can be applied
* If there are manual Groups in code config, Grouparoo will fail to validate/apply/start/run with an error message
* If there are calculated Groups in code config, Grouparoo will boot, and provide a deprecation warning that `group.type` is no longer a property of a Grouparoo group.

Docs @ https://github.com/grouparoo/www.grouparoo.com/pull/828

### Development

- [ ] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: core/src/models/GroupRule.ts [hunk 1]
```diff
@@ -37,17 +43,17 @@ export class GroupRule extends CommonModel<GroupRule> {
...
 
-  @Column
-  relativeMatchUnit: string;
+  @Column(DataType.STRING)
+  relativeMatchUnit: RelativeMatchUnitType;
 
-  @Column
-  relativeMatchDirection: string;
+  @Column(DataType.STRING)
+  relativeMatchDirection: RelativeMatchDirectionType;
```
Comment (evantahler) [inline core/src/models/GroupRule.ts:56]: Adds better TS Types

File: core/src/models/Group.ts [hunk 2]
```diff
@@ -50,14 +55,13 @@ export interface GroupRuleWithKey {
...
   type?: string;
   topLevel?: boolean;
-  operation: { op: string; description?: string };
+  operation: { op: GroupRuleOpType; description?: string };
   match?: string | number | boolean;
   relativeMatchNumber?: number;
-  relativeMatchUnit?: string;
-  relativeMatchDirection?: string;
+  relativeMatchUnit?: RelativeMatchUnitType;
+  relativeMatchDirection?: RelativeMatchDirectionType;
```
Comment (evantahler) [inline core/src/models/Group.ts:62]: Adds better TS Types

File: core/src/modules/configLoaders/group.ts [hunk 3]
```diff
@@ -11,14 +11,15 @@ import { Group } from "../../models/Group";
...
+import { Deprecation } from "../deprecation";
 
 export async function loadGroup(
   configObject: GroupConfigurationObject,
   externallyValidate: boolean,
   validate = false
 ): Promise<IdsByClass> {
   let isNew = false;
-  validateConfigObjectKeys(Group, configObject, ["rules"]);
+  validateConfigObjectKeys(Group, configObject, ["rules", "type"]);
```
Comment (evantahler) [inline core/src/modules/configLoaders/group.ts:22]: For now, we won't crash if `group.type` is provided on a config object, but we check it below

File: core/src/modules/ruleOpsDictionary.ts [hunk 4]
```diff
@@ -85,3 +99,12 @@ export function buildPropertyOpsDictionary(cfg?: any) {
...
 export const PropertyOpsDictionary = buildPropertyOpsDictionary(config || {});
+
+export type GroupRuleOpType =
+  | typeof _boolean_ops[number]["op"]
+  | typeof _number_ops[number]["op"]
+  | typeof _string_ops_sqlite[number]["op"]
+  | typeof _string_ops_postgres[number]["op"]
+  | typeof _date_ops[number]["op"];
+export type RelativeMatchUnitType = typeof _relativeMatchUnits[number];
+export type RelativeMatchDirectionType = "add" | "subtract";
```
Comment (evantahler) [inline core/src/modules/ruleOpsDictionary.ts:110]: Finally providing types for the Group Rules

File: core/__tests__/integration/happyPath.ts [hunk 5]
```diff
@@ -259,90 +255,10 @@ describe("integration/happyPath", () => {
...
-      expect(simpleRecordValues(records[0].properties).email).toEqual([
-        "luigi@example.com",
-      ]);
-    });
-  });
-
-  describe("calculated group", () => {
-    let groupId: string;
-
     test("a calculated group can be created", async () => {
```
Comment (teallarson) [inline core/__tests__/integration/happyPath.ts]:
```suggestion
    test("a group can be created", async () => {
```

**Review by pedroslopez — APPROVED** [review]
This is a big one! Looks good though and I think the migration path is clear.

**Review by teallarson — APPROVED** [review]
Bye Manual Groups!  I found one spot that still used "Calculated Group" instead of "Group" in a test, but other than that looks good.

**Pull request merged**